src/ansys/pyensight/core/listobj.py

```diff
@@ -141,13 +141,17 @@ def set_attr(self, attr: Any, value: Any) -> int:
 
         """
         count: int = 0
+        objid_list = []
+        session = None
+        objid_list = [x.__OBJID__ for x in self if isinstance(x, ENSOBJ)]
+        count = len(objid_list)
         for item in self:
-            if isinstance(item, ENSOBJ):
-                try:
-                    item.setattr(attr, value)
-                    count += 1
-                except RuntimeError:
-                    pass
+            if hasattr(item, "_session"):
+                session = item._session
+                break
+        if session:
```

Comment:
@mariostieriansys The return value of set_attrs is not the number of items in the input list, rather the number of items in the input list that could successfully have the values set.   I believe the return value should be:

count = session.cmd(msg)

Consider a mixed list of annots where some support the attribute and some do not.

Reply:
makes sense, I am fixing it